trustd/src/main.rs

```diff
@@ -25,119 +23,52 @@ pub enum Command {}
                     if n == 0 {
-                        log::error!("Caused by:");
+                        eprintln!("Caused by:");
                     }
-                    log::error!("\t{err}");
+                    eprintln!("\t{err}");
                 }
 
                 ExitCode::FAILURE
             }
         }
     }
 
-    async fn run_command(mut self) -> anyhow::Result<ExitCode> {
-        // to keep in scope while running.
-        let mut managed_db = None;
-
-        if matches!(self.database.db_strategy, DbStrategy::Managed) {
-            println!("setting up managed DB");
-            use postgresql_embedded::Settings;
-
-            let current_dir = env::current_dir()?;
-            let work_dir = current_dir.join(".trustify");
-            let db_dir = work_dir.join("postgres");
-            let settings = Settings {
-                username: self.database.username.clone(),
-                password: self.database.password.clone(),
-                temporary: false,
-                installation_dir: db_dir.clone(),
-                ..Default::default()
-            };
-
-            let mut postgresql = PostgreSQL::new(PostgreSQL::default_version(), settings);
-            postgresql.setup().await?;
-            postgresql.start().await?;
-
-            let port = postgresql.settings().port;
-            self.database.port = port;
-            self.creation = CreationMode::Bootstrap;
-
-            managed_db.replace(postgresql);
-
-            println!("postgresql installed under {:?}", db_dir);
-            println!("running on port {}", port);
-        }
-
-        let mut handles = JoinSet::new();
-
-        if self.with_http {
-            let http = trustify_server::Run {
-                database: self.database.clone(),
-                storage: self.storage.clone(),
-                creation: self.creation,
-                devmode: self.devmode,
-                infra: self.infra.clone(),
-                auth: self.auth.clone(),
-                http: self.http.clone(),
-                swagger_ui_oidc: self.swagger_ui_oidc.clone(),
-            };
-
-            handles.spawn_local(http.run());
-        }
-
-        while let Some(result) = handles.join_next().await {
-            match result {
-                Ok(result) => match result {
-                    Ok(_) => {}
-                    Err(err) => {
-                        log::error!("error {:?}", err);
-                    }
-                },
-                Err(err) => {
-                    log::error!("fundamental error {:?}", err);
+    async fn run_command(self) -> anyhow::Result<ExitCode> {
+        match self.command {
+            Some(Command::Api(run)) => run.run().await,
+            Some(Command::Db(run)) => run.run().await,
+            None => {
+                let Some(Command::Db(mut db)) =
+                    Trustd::parse_from(["trustd", "db", "migrate"]).command
```

Comment:
So we're taking "no arguments" (and only the no-arg case?) and re-feeding it back in implying a set of arguments?

Reply:
Yes, `None` => PM mode. In fact, that's a good name for a fn to put that in.